channeldb/payments.go

```diff
@@ -116,6 +117,50 @@ func (db *DB) FetchAllPayments() ([]*OutgoingPayment, error) {
 	return payments, nil
 }
 
+// LookupPayment attempts to look up an payment according to its 32 byte
+// payment hash.
+func (db *DB) LookupPayment(paymentHash [32]byte) (*OutgoingPayment, error) {
+	var payment *OutgoingPayment
+
+	err := db.View(func(tx *bolt.Tx) error {
```

Comment:
Could you break this expression up into helper functions? Having a function within a function within a function makes this a bit hard to read, along with the fact that it doesn't all fit on one screen. I imagine something like the following:
`
func (db *DB) LookupPayment(...) ... {
    ...
        db.View(processPayments)
    ...
}

func processPayments(...) (OutgoingPayment, error) {
    ...
    var payment *OutgoingPayment
    var err error
    payments.ForEach(func(k, v []byte) error {
        tempPayment, err := processPayment(v)
        if err != nil {
            return err
        }
        payment = tempPayment
    })

    return payment, err
}

func processPayment(...) (OutgoingPayment, error) {
    ...
    return payment, err
}
`

Breaking code down into smaller chunks can make it more palatable for the reader and easier to write tests for.

Also, there's probably better names for the helper functions than "processPayments" and "processPayment".